assignment/tool/src/webapp/js/assignmentsByStudent.js

```diff
@@ -55,21 +57,27 @@ function parseDataCell(html) {
   };
 }
 
-function renderGrouping({ studentName, actionLink, expanded }) {
+function renderGrouping({ studentName, actionLink, expanded, studentUserId }) {
   const template = document.createElement('template');
   template.innerHTML = `
     <tr>
       <td class="border-0">
-        <a href="${actionLink}">
+        <a href="${actionLink}" id="${studentUserId}">
           <span class="expand-icon si ${expanded ? "si-expanded" : "si-collapsed"}"
               aria-hidden="true"></span>
           <span>${studentName}</span>
         </a>
       </td>
     </tr>
   `;
-  
+  if (!itChanged && currentExpandedUserId != "") {
```

Comment:
When moving this to the vm in the load callback, you could also the velocity `#if` and you don't need to track this in the js.